src/provisioning/mod.rs

```diff
@@ -85,6 +86,10 @@ pub enum ProvisioningError {
     EncryptOnlyProvisioningCipher,
     #[error("invalid profile key bytes")]
     InvalidProfileKey(TryFromSliceError),
+    #[error(transparent)]
+    Curve(#[from] CurveError),
+    #[error(transparent)]
+    Base64(#[from] base64::DecodeError),
```

Comment:
Should we consolidate these three errors? I don't think we should expose low-level errors like Curve and base64 in the public API.